java-frontend/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -86,7 +87,21 @@ public String valueAsString() {
   private static final String JAVA_IO_AUTO_CLOSEABLE = "java.lang.AutoCloseable";
   private static final String JAVA_IO_CLOSEABLE = "java.io.Closeable";
   private static final String JAVA_SQL_STATEMENT = "java.sql.Statement";
-  private static final String[] JDBC_RESOURCE_CREATIONS = {"java.sql.Connection", JAVA_SQL_STATEMENT};
+  private static final String JAVA_SQL_CONNECTION = "java.sql.Connection";
+  private static final String JAVAX_SQL_DATA_SOURCE = "javax.sql.DataSource";
```

Comment:
Not sure you need dedicated constants for `javax.sql.DataSource` and `java.sql.PreparedStatement`, as they are used only once.